### ISSUE TITLE
bug: java.util.concurrent.TimeoutException

### ISSUE THREAD

**Issue by RuixiangS** [sourcegraph/jetbrains#879]
Plugin version: ```5.3.1544```
IDE version: ```RR-233.14015.153```
Stacktrace:
```text
java.util.concurrent.TimeoutException
	at java.base/java.util.concurrent.CompletableFuture.timedGet(CompletableFuture.java:1960)
	at java.base/java.util.concurrent.CompletableFuture.get(CompletableFuture.java:2095)
	at com.sourcegraph.cody.agent.CodyAgentService.startAgent$lambda$3(CodyAgentService.kt:69)
	at com.intellij.openapi.application.impl.ApplicationImpl$2.run(ApplicationImpl.java:249)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
	at com.intellij.util.concurrency.ContextCallable.call(ContextCallable.java:32)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at com.intellij.util.concurrency.ContextRunnable.run(ContextRunnable.java:27)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:702)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:699)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1.run(Executors.java:699)
	at java.base/java.lang.Thread.run(Thread.java:840)

```

**Comment by mkondratek** [comment]
hello @RuixiangS 👋 thank you for posting this issue and I am sorry you experienced that.

We assume that this error is related to the internet connection - as a quick workaround you can try switch to better connection. We also merged a few related fixes & improvements recently (in particular: https://github.com/sourcegraph/jetbrains/pull/756) - the issue should be fixed there. 

The plugin version with the fix as now available as a pre-release version [v5.3.2625-nightly](https://github.com/sourcegraph/jetbrains/releases/tag/v5.3.2625) (JetBrains Marketplace link: [5.3.2625-nightly](https://plugins.jetbrains.com/plugin/9682-sourcegraph-cody--code-search/versions/nightly/495190)) and most likely it will be released as a release version (stable channel version: **5.3.2625**) next week. 

I kindly ask you for some patience 🙏

**Comment by pkukielka** [comment]
I'm closing this in favour of https://github.com/sourcegraph/jetbrains/issues/723
Also situation should be greatly improved in [5.3.2999-nightly](https://plugins.jetbrains.com/plugin/9682-sourcegraph-cody--code-search/versions/nightly/496995) release which should become available as stable today.
